plugin/nfs/nfs.py

```diff
@@ -0,0 +1,571 @@
+
+        return
+
+    def time_out_set(self, ms, flags=0):
+        self.tmo = ms
+
+    def time_out_get(self, flags=0):
+        return self.tmo
+
+    @staticmethod
+    def _get_fs_sizes(path):
+        """Stat a filesystem and return total and available space"""
+        try:
+            st = os.statvfs(path)
+            total_size = st.f_blocks * st.f_frsize
+            avail_size = st.f_bavail * st.f_frsize
+        except OSError:
+            raise
+
+        return (total_size, avail_size)
+
+    def pools(self, search_key=None, search_value=None, flags=0):
+        pools = []
+        parts = list_mounts()
+        for p in parts:
+            try:
+                (total_size, avail_size) = NFSPlugin._get_fs_sizes(p)
+                fsid = get_fsid(p)
+                pooltype = Pool.ELEMENT_TYPE_FS
+                unsup_actions = 0
+                status = System.STATUS_OK
+                status_info = ''
+                pools.append(Pool(fsid, p, pooltype, unsup_actions,
+                                  total_size, avail_size, status, status_info,
+                                  self._SYSID))
+            except OSError:
+                pass
+        return search_property(pools, search_key, search_value)
+
+    def systems(self, flags=0):
+        syslist = []
+        hostname = os.uname()[1]
+        syslist.append(System(NFSPlugin._SYSID, hostname,
+                       System.STATUS_UNKNOWN, ''))
+        return syslist
+
+    def fs(self, search_key=None, search_value=None, flags=0):
+        """List filesystems, required for other operations"""
+        fss = []
+        parts = list_mounts()
+        for p in parts:
+            try:
+                (total_size, avail_size) = NFSPlugin._get_fs_sizes(p)
+                fsid = get_fsid(p)
+                fss.append(
+                    FileSystem(fsid, p, total_size,
+                               avail_size, fsid, self._SYSID))
+            except OSError:
+                pass
+
+        return search_property(fss, search_key, search_value)
+
+    def exports(self, search_key=None, search_value=None, flags=0):
+        """List existing exports"""
+        exports = NFSPlugin._load_exports()
+        return search_property(exports, search_key, search_value)
+
+    def export_fs(self, fs_id, export_path, root_list, rw_list, ro_list,
+                  anon_uid=NfsExport.ANON_UID_GID_NA,
+                  anon_gid=NfsExport.ANON_UID_GID_NA,
+                  auth_type=None, options=None, flags=None):
+        """Add an export"""
+
+        if fs_id is None and export_path is None:
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           'Must provide fs_id or export_path')
+
+        if fs_id is None:
+            fs_id = get_fsid(export_path)
+            if fs_id is None:
+                raise LsmError(ErrorNumber.NOT_FOUND_FS,
+                               'FileSystem not found')
```

Comment:
Need to verify file system is exists when `fs_id` is not `None`, or raise `NOT_FOUND_FS`.